crates/cli-support/src/anyref.rs

```diff
@@ -45,13 +47,51 @@ pub fn process(module: &mut Module) -> Result<(), Error> {
 
     let meta = cfg.run(module)?;
 
-    let section = module
+    // Store functions found during the anyref transformation onto the auxiliary
+    // section. These will end up getting used for the full support in the JS
+    // polyfill, so we'll want the functions to stick around.
+    let mut aux = module
         .customs
-        .get_typed_mut::<WasmBindgenAux>()
+        .delete_typed::<WasmBindgenAux>()
         .expect("wit custom section should exist");
-    section.anyref_table = Some(meta.table);
-    section.anyref_alloc = meta.alloc;
-    section.anyref_drop_slice = meta.drop_slice;
+    aux.anyref_table = Some(meta.table);
+    aux.anyref_alloc = meta.alloc;
+    aux.anyref_drop_slice = meta.drop_slice;
+
+    // Additionally if the heap live count intrinsic was found, if it was
+    // actually called from any adapter function we want to switch to calling
+    // the wasm core module intrinsic directly. This'll avoid generating
+    // incorrect JS glue.
+    if let Some(id) = meta.live_count {
+        let section = module
+            .customs
+            .get_typed_mut::<NonstandardWitSection>()
+            .expect("wit custom section should exist");
+        for (_, adapter) in section.adapters.iter_mut() {
+            let instrs = match &mut adapter.kind {
+                AdapterKind::Local { instructions } => instructions,
+                AdapterKind::Import { .. } => continue,
+            };
+            for instr in instrs {
+                let adapter = match instr.instr {
+                    Instruction::CallAdapter(id) => id,
+                    _ => continue,
+                };
+                let import = match aux.import_map.get(&adapter) {
+                    Some(import) => import,
+                    None => continue,
+                };
+                match import {
+                    AuxImport::Intrinsic(Intrinsic::AnyrefHeapLiveCount) => {}
```

Comment:
These nested loops are really calling for some kind of traversal / walker or something with filtering support for adapter instructions...